server/src/main/java/org/opensearch/cluster/awarenesshealth/ClusterAwarenessAttributeValueHealth.java

```diff
@@ -0,0 +1,345 @@
+                        weight = parser.doubleValue();
+                        break;
+                }
+            } else {
+                throw new OpenSearchParseException(
+                    "failed to parse awareness attribute health, expected [{}] but found [{}]",
+                    XContentParser.Token.FIELD_NAME,
+                    token
+                );
+            }
+        }
+        return new ClusterAwarenessAttributeValueHealth(
+            attributeName,
+            active_shards,
+            initializing_shards,
+            relocating_shards,
+            unassigned_shards,
+            nodes,
+            weight
+        );
+    }
+
+    public int getActiveShards() {
+        return activeShards;
+    }
+
+    public void setActiveShards(int activeShards) {
+        this.activeShards = activeShards;
+    }
+
+    public int getUnassignedShards() {
+        return unassignedShards;
+    }
+
+    public void setUnassignedShards(int unassignedShards) {
+        this.unassignedShards = unassignedShards;
+    }
+
+    public int getNodes() {
+        return nodes;
+    }
+
+    public void setNodes(int nodes) {
+        this.nodes = nodes;
+    }
+
+    public double getWeight() {
+        return weight;
+    }
+
+    public void setWeight(double weight) {
+        this.weight = weight;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public int getInitializingShards() {
+        return initializingShards;
+    }
+
+    public void setInitializingShards(int initializingShards) {
+        this.initializingShards = initializingShards;
+    }
+
+    public int getRelocatingShards() {
+        return relocatingShards;
+    }
+
+    public void setRelocatingShards(int relocatingShards) {
+        this.relocatingShards = relocatingShards;
+    }
+
+    public List<String> getNodeList() {
+        return nodeList;
+    }
+
+    public void setNodeList(List<String> nodeList) {
+        this.nodeList = nodeList;
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeString(name);
+        out.writeVInt(activeShards);
+        out.writeVInt(initializingShards);
+        out.writeVInt(relocatingShards);
+        out.writeVInt(unassignedShards);
+        out.writeVInt(nodes);
+        out.writeDouble(weight);
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject(getName());
+        builder.field(ACTIVE_SHARDS, getActiveShards());
+        builder.field(INITIALIZING_SHARDS, getInitializingShards());
+        builder.field(RELOCATING_SHARDS, getRelocatingShards());
+        builder.field(UNASSIGNED_SHARDS, getUnassignedShards());
+        builder.field(NODES, getNodes());
+        builder.field(WEIGHTS, getWeight());
+        builder.endObject();
+        return builder;
+    }
+
+    protected void computeAttributeValueLevelInfo(
```

Comment:
Make this package private instead? Why protected?

Reply:
This is used by ClusterAwarenessAttributesHealth to call per attribute value which is a different class and thus not making this private